File: Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(int A[], int n) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        int result = INT_MIN, last = -1;
        for (int i = 0; i < n; i++) {
            if (last <= 0) {
                last = A[i];
            } else {
                last += A[i];
            }
            result = max(result, last);
        }
        return result;
    }
};

File: Maximum Subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        var result = int.MinValue;
        var last = -1;
        for (var i = 0; i < nums.Length; i++) {
            if (last <= 0) {
                last = nums[i];
            } else {
                last += nums[i];
            }
            result = Math.Max(result, last);
        }
        
        return result;
    }
}
